resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/providers/sse/SseEventInputImpl.java

```diff
@@ -212,25 +214,40 @@ else if ("retry".equals(name))
       EventByteArrayOutputStream buffer = new EventByteArrayOutputStream();
       int data;
       int pos = 0;
+      boolean boundary = false;
+      byte[] eolBuffer = new byte[5];
       while ((data = in.read()) != -1)
```

Comment:
@jimma, at the end of the loop, you return null. What if a line is terminated by EOF? The BNF says
```
end-of-line   = ( cr lf / cr / lf / eof )
```

Reply:
@ronsigal From the latest spec https://html.spec.whatwg.org/multipage/server-sent-events.html#server-sent-events, I only see end-of-line =  ( cr lf / cr / lf )

Reply:
@jimma, ah, I was looking at https://www.w3.org/TR/2009/WD-eventsource-20091029/#parsing-an-event-stream.